src/memusage.h

```diff
@@ -16,13 +16,59 @@
 #include <unordered_map>
 #include <unordered_set>
 
-
 namespace memusage
 {
 
 /** Compute the total memory used by allocating alloc bytes. */
 static size_t MallocUsage(size_t alloc);
 
+/** Wrapping allocator that accounts accurately.
+ *
+ * To use accounting, the AccountAllocator(size_t&) constructor must be used.
+ * Any containers using such an allocator, and containers copied/moved from such
+ * containers will then increment/decrement size_t when allocating/deallocating
+ * memory.
+ */
+template<typename T>
+class AccountingAllocator : public std::allocator<T>
+{
+    size_t* m_allocated; //!< Pointer to accounting variable, cannot be nullptr
```

Comment:
Shouldn't be documented here that the accounting variable must be atomic or protected by mutex?

Reply:
I've added a bunch of comments.